R/klass.R

```diff
@@ -0,0 +1,146 @@
+#' Get municipalities
+#'
+#' Get all Norwegian municipalities and municipality codes for a given year.
+#'
+#' Fetches all Norwegian municipalities and municipality codes for a given year.
+#' All years from 1977 until present are supported. The function will default to
+#' the current year. The function will return a data.frame with all municipalities.
+#' If more than one year is given, a single data.frame will be returned with all
+#' codes and names for the given years, unless `simplify` is set to `FALSE`, then
+#' a list of the same length of the input will be returned instead.
+#'
+#' You can add county numbers and names for each municipality by setting `add_county`
+#' to `TRUE`.
+#'
+#' @inheritParams get_entity
+#' @param year The year for which the codes should be valid for
+#' @param add_county If set to `TRUE`, add county name and number
+#' @param include_notes If notes should be included or not
+#' @param simplify If `TRUE` (default), a single data.frame is returned.
+#'
+#' @return data.frame or list
+#'
+#' @seealso `get_counties()` for county codes and names
+#'
+#' @export
+get_municipalities <- function(
+    year = format(Sys.Date(), '%Y'), add_county = FALSE, include_notes = FALSE,
+    simplify = TRUE, raw_response = FALSE)
+{
+  if (raw_response) simplify <- FALSE
+  if (add_county) {
+    raw_response <- FALSE
```

Comment:
I'm not too crazy about chaging input parameters "behind the scenes" . If we do this, we should at least warn the user. But more importantly I don't think `add_county` and `raw_response` need to be mutally exclusive.

Reply:
The best solution (for now) is to issue a warning to the user that they have given incompatible input arguments, and that `raw_response` will be ignored. I do agree with the principle you are raising, but the cost-benefit of rewriting the function to support both scenarios is probably low. This can be added as a new issue for a future release.

Reply:
@ohjakobsen: As discussed I agree that a warning is okay for now.

Reply:
See comment above.